src/js/model/Score.js

```diff
@@ -0,0 +1,71 @@
+function Score() {
+  this._scoreboard = [];
+  this._total = 0;
+
+  this._rolls = [];
+}
+
+Score.prototype.getScoreBoard = function() {
+  return this._scoreboard;
+};
+
+Score.prototype.getScoreTotal = function() {
+  return this._total;
+};
+
+Score.prototype.getRolls = function() {
+  return this._rolls;
+};
+
+Score.prototype.rollNumber = function() {
+  return this.getRolls().length;
+};
+
+Score.prototype.rollMinusOne = function() {
+  return this.getRolls().slice(-2)[0];
+};
+
+Score.prototype.rollMinusTwo = function() {
+  return this.getRolls().slice(-3)[0];
+}
+
+Score.prototype.addBonus = function(roll) {
+  this._scoreboard[this._scoreboard.length - 2] += roll;
+}
+
+Score.prototype.recordRolls = function(roll) {
+  this._rolls.push(roll);
+};
+
+Score.prototype.setScore = function(roll, last_frame, this_frame, game_length) {
+  this.recordRolls(roll);
+  this.scorePins(roll, this_frame);
+  this.scoreSpare(roll, this_frame, last_frame);
+  this.scoreStrike(roll, game_length);
+  this._total = this._scoreboard.reduce(function(a,b) { return a + b });
+};
+
+Score.prototype.scorePins = function(roll, current_frame) {
+  if (current_frame.isNew()) { this._scoreboard.push(roll); }
+  else { this._scoreboard[this._scoreboard.length - 1] += roll; }
+}
+
+Score.prototype.scoreSpare = function(roll, current_frame, last_frame) {
+  if (typeof last_frame !== 'undefined') {
+    if (last_frame.isSpare() && current_frame.isNew()) { this._scoreboard[this._scoreboard.length - 2] += roll;; }
```

Comment:
The double semicolon! Extra safe. Consider dropping the habit though — maybe `jshint` would help by nagging you out of it?

I'd also consider putting a few more linebreaks into the ifs/functions in the code. It feels kind of 'neat' to do oneliners, but it's usually much more readable to have stuff visually broken up. Here's how I'd write it:

```js
Score.prototype.scoreSpare = function(roll, current_frame, last_frame) {
  if (typeof last_frame == 'undefined') { return; }

  if (last_frame.isSpare() && current_frame.isNew()) {
    this._scoreboard[this._scoreboard.length - 2] += roll;
  }
};
```

If the body of the if is just `return`, making it a guard clause, you can just about get away with it — and some JS devs won't even like that.